src/adu_workflow/src/agent_workflow.c

```diff
@@ -555,7 +556,17 @@ void ADUC_Workflow_HandlePropertyUpdate(
     else
     {
         // This is a top level workflow, make sure that we set the working folder correctly.
-        workflow_set_workfolder(nextWorkflow, "%s/%s", ADUC_DOWNLOADS_FOLDER, workflow_peek_id(nextWorkflow));
+        const ADUC_ConfigInfo* config = ADUC_ConfigInfo_GetInstance();
+        if (config != NULL)
```

Comment:
nit: would be clean to add a ConfigIsNotNUllOrEmpty() function to make these checks more aggressive and cleaner.